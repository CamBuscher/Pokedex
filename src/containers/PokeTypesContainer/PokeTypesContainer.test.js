import React from 'react';
import { shallow } from 'enzyme';
import { connect } from 'react-redux'

import { PokeTypesContainer, mapDispatchToProps, mapStateToProps } from './index';

describe('PokeTypesContainer', () => {
  let mockProps
  let wrapper

  beforeEach(() => {
    mockProps = {
      getPokeTypes: jest.fn(),
      isLoading: false,
      hasErrored: false,
      types: []
    } 

    wrapper = shallow(<PokeTypesContainer {...mockProps} />)
  })

  it('should match snapshot when loading is false', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot when loading is true', () => {
    mockProps = {
      getPokeTypes: jest.fn(),
      isLoading: true,
      hasErrored: false,
      types: []
    }

    wrapper = shallow(<PokeTypesContainer {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
  })
})