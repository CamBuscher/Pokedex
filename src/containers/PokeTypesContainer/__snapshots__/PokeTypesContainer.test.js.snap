// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeTypesContainer should match snapshot when loading is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PokeTypesContainer
    getPokeTypes={[Function]}
    isLoading={false}
    types={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "pokeTypesContainer",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "pokeTypesContainer",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PokeTypesContainer should match snapshot when loading is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PokeTypesContainer
    getPokeTypes={[Function]}
    isLoading={true}
    types={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        src="https://media.giphy.com/media/slVWEctHZKvWU/giphy.gif"
/>,
      "className": "pokeTypesContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "src": "https://media.giphy.com/media/slVWEctHZKvWU/giphy.gif",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          src="https://media.giphy.com/media/slVWEctHZKvWU/giphy.gif"
/>,
        "className": "pokeTypesContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "https://media.giphy.com/media/slVWEctHZKvWU/giphy.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
